sphinx_search/extension.py

```diff
@@ -24,13 +31,24 @@ def copy_asset_files(app, exception):
 
 
 def inject_static_files(app):
-    """Inject correct CSS and JS files based on the value of ``RTD_SPHINX_SEARCH_FILE_TYPE``."""
+    """
+    Inject correct CSS and JS files based on the value of ``RTD_SPHINX_SEARCH_FILE_TYPE``.
+
+    This only injects file if the docs are build on Read the Docs.
+    """
+
+    on_rtd = os.environ.get('READTHEDOCS') == 'True'
+
+    # only inject files if the builder is one of the ONLINE_BUILDERS
+    # and on_rtd is True
+    if app.builder.name not in ONLINE_BUILDERS and not on_rtd:
```

Comment:
Instead of changing the logic here, we should make the code log something and not activate when not running on RTD.

Reply:
I don't follow.
What do you mean by `not activate when not running on RTD.`?
Did you mean something like this: https://github.com/readthedocs/readthedocs-sphinx-search/blob/34afc8bf5246553d651bb0f14f6a7260a311ed32/sphinx_search/_static/js/rtd_sphinx_search.js#L665

Reply:
Correct. Since the API hostname comes out of `READTHEDOCS_DATA`, if you can't get the API host just don't do anything. Don't do any overlay and just fallback to regular sphinx in-doc search.